scipy/special/_logsumexp.py

```diff
@@ -115,7 +115,10 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
         s = np.sum(tmp, axis=axis, keepdims=keepdims)
         if return_sign:
             sgn = np.sign(s)
-            s *= sgn  # /= makes more sense but we need zero -> zero
+            if np.issubdtype(sgn.dtype, np.floating):
+                s *= sgn  # /= makes more sense but we need zero -> zero
+            else:
+                s /= np.where(sgn == 0, 1, sgn)
```

Comment:
Since this works also for `floating`, does it make more sense to just use it for all cases?

Reply:
I didn't do that because it makes the implementation more expensive in the case that already works correctly. Would you prefer erring toward simplicitly of implementation?

Reply:
If we were opting to always use the numpy 2.0 definition of complex sign, a simpler option would be `s = abs(s)`.

Reply:
Indeed, `s = abs(s)` would make the intent much clearer.... 

Since a logarithm is taken, I think the division does not matter too much. But a fast alternative would be
```
s *= sgn.conjugate()
```
which would I think be consistent for all cases too. 

But perhaps we're overthinking!?

Reply:
Fundamentally, I think the question is whether we want the sign here to be `np.sign(x)`, which changes depending on numpy version, or `x / abs(x)` in all cases.

Reply:
Ah, yes, it might make sense to just define it usefully for all numpy versions. But I should leave that to the scipy maintainers!